src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/YFlowCrudSpec.groovy

```diff
@@ -36,122 +51,498 @@ class YFlowCrudSpec extends HealthCheckSpecification {
     @Autowired @Shared
     Provider<TraffExamService> traffExamProvider
 
-    @Ignore
     @Tidy
     @Tags([TOPOLOGY_DEPENDENT])
-    @Unroll("Valid #data.description has traffic and no rule discrepancies \
-(#flow.source.switchId - #flow.destination.switchId)")
-    def "Valid y-flow has no rule discrepancies"() {
-        given: "A y-flow"
-        assumeTrue(topology.activeTraffGens.size() >= 2,
-"There should be at least two active traffgens for test execution")
-        def traffExam = traffExamProvider.get()
-        def allLinksInfoBefore = northbound.getAllLinks().collectEntries { [it.id, it.availableBandwidth] }.sort()
-        flowHelperV2.addFlow(flow)
-        def path = PathHelper.convert(northbound.getFlowPath(flow.flowId))
-        def switches = pathHelper.getInvolvedSwitches(path)
-        //for single-flow cases need to add switch manually here, since PathHelper.convert will return an empty path
-        if (flow.source.switchId == flow.destination.switchId) {
-            switches << topology.activeSwitches.find { it.dpId == flow.source.switchId }
+    def "Valid y-flow can be created#trafficDisclaimer, covered cases: #coveredCases"() {
+        assumeTrue(swT != null, "These cases cannot be covered on given topology: $coveredCases")
+
+        when: "Create a y-flow of certain configuration"
+        def yFlow = northboundV2.addYFlow(yFlowRequest)
+
+        then: "Y-flow is created and has UP status"
+        Wrappers.wait(WAIT_OFFSET) { assert northboundV2.getYFlow(yFlow.YFlowId).status == FlowState.UP.toString() }
+
+        and: "2 sub-flows are created, visible via regular 'dump flows' API"
+        northboundV2.getAllFlows()*.flowId.sort() == yFlow.subFlows*.flowId.sort()
```

Comment:
is it visible for both versions of API (dumpFlowsV1/dumpFlowsV2)?

Reply:
yes, checked it manually. Don't see any point in adding tests for v1, since they use the same backend